src/rustup/config.rs

```diff
@@ -93,7 +107,8 @@ impl Cfg {
             gpg_key: gpg_key,
             notify_handler: notify_handler,
             env_override: env_override,
-            dist_root_url: dist_root_url,
+            dist_root_url: dist_root,
```

Comment:
@brson I am afraid not.

The `dist_root_url` field is accessed in `rustup::toolchain`(see [here](https://github.com/rust-lang-nursery/rustup.rs/blob/master/src/rustup/toolchain.rs#L146)), which is used to construct the `DownloadCfg`.

Because the type of DownloadCfg.dist_root is `&str`, it is impossible to assign a modified value.